packages/language-server/src/lib/MessageHandler.ts

```diff
@@ -79,14 +79,18 @@ export function handleDiagnosticsRequest(
   }
 
   for (const diag of res) {
+    const previewNotKnownRegex = /The preview feature \"[a-zA-Z]+\" is not known/
     const diagnostic: Diagnostic = {
       range: {
         start: document.positionAt(diag.start),
         end: document.positionAt(diag.end),
       },
-      message: diag.text,
-      source: '',
+      message: previewNotKnownRegex.test(diag.text)
```

Comment:
Would there be a way to use metadata somehow instead of a regex?

Reply:
Further context from DM:
> So I don't know much of the codebase, but I was wondering if instead of testing via previewNotKnownRegex.test(diag.text), we could get the same information in another/cleaner way (eg. via an object passed down, or a property in diag). (@millsp)

My response:
> So, unfortunately, we do not use the LSP diagnostic type in engines and perform a conversion in LT. This is all of the info we get from engines currently
![image](https://github.com/prisma/language-tools/assets/29753584/bc7cf1f3-4f45-4689-b441-4ef0f7701bc3)
so we need to scan text unless we want to update engines as well